src/proxy/app/db_manager.py

```diff
@@ -1,61 +1,149 @@
-""" Database manager """
+"""Database manager"""
 
+import asyncio
 import logging
+from datetime import datetime
 
 import asyncpg
-from fastapi import FastAPI, HTTPException
+from azure.identity import DefaultAzureCredential
+from fastapi import HTTPException
 
-logging.basicConfig(level=logging.WARNING)
+# 6 hr recycle time
+RECYCLE_TIME_SECONDS = 60 * 60 * 6
+
+logging.basicConfig(level=logging.INFO)
+
+
+class DBConfig:
+    """Database configuration"""
+
+    def __init__(
+        self,
+        host: str,
+        port: int,
+        database: str,
+        user: str,
+        password: str,
+        encryption_key: str,
+        connection_string: str,
+    ) -> None:
+        self.host = host.strip() if host else None
+        self.port = port
+        self.database = database.strip() if database else None
+        self.user = user.strip() if user else None
+        self.password = password.strip() if password else None
+        self.encryption_key = encryption_key.strip() if encryption_key else None
+        self.connection_string = connection_string.strip() if connection_string else None
+        self.logging = logging.getLogger(__name__)
+
+        if not connection_string:
+            if not host or not user:
+                raise HTTPException(
+                    status_code=500,
+                    detail="Please set the environment variables POSTGRES_SERVER, POSTGRES_USER",
+                )
+
+        if not encryption_key:
+            raise HTTPException(
+                status_code=500,
+                detail="Please set the environment variable POSTGRES_ENCRYPTION_KEY",
+            )
+
+    def get_connection_string(self):
+        """get connection string"""
+        if self.connection_string:
+            return self.connection_string
+
+        if not self.password:
+            self.logging.info("Using Postgres Entra Token Authorization")
+            azure_credential = DefaultAzureCredential()
+            token = azure_credential.get_token(
+                "https://ossrdbms-aad.database.windows.net/.default"
+            ).token
+
+            connection_string = (
+                f"postgresql://{self.user}:{token}@{self.host}:{self.port}/{self.database}"
```

Comment:
Interesting, you dont need to set sslmode? We sometimes find that we do need to set it for Azure PostgreSQL Flexible.